Intersect.Client/Entities/Entity.cs

```diff
@@ -1016,69 +1014,90 @@ public virtual void Draw()
+            var renderColor = new Color(Color.A, Color.R, Color.G, Color.B);
+
+            var source = AnimatedTextures[SpriteAnimation] ?? Texture;
+            var dest = TextureCache[animation];
+
+            //Order the layers of paperdolls and sprites
+            for (var z = 0; z < Options.PaperdollOrder[Dir].Count; z++)
+            {
+                var paperdoll = Options.PaperdollOrder[Dir][z];
+                var equipSlot = Options.EquipmentSlots.IndexOf(paperdoll);
+
+                //Check for player
+                if (paperdoll == "Player")
+                {
+                    Graphics.DrawGameTexture(source, 0, 0, Color, dest);
+                }
+                else if (equipSlot > -1)
+                {
+                    //Don't render the paperdolls if they have transformed.
+                    if (!IsTransformed && Equipment.Length == Options.EquipmentSlots.Count)
                     {
-                        //Don't render the paperdolls if they have transformed.
-                        if (sprite == Sprite && Equipment.Length == Options.EquipmentSlots.Count)
+                        if (Equipment[equipSlot] != Guid.Empty && this != Globals.Me ||
+                            MyEquipment[equipSlot] < Options.MaxInvItems)
                         {
-                            if (Equipment[equipSlot] != Guid.Empty && this != Globals.Me ||
-                                MyEquipment[equipSlot] < Options.MaxInvItems)
+                            var itemId = Guid.Empty;
+                            if (this == Globals.Me)
                             {
-                                var itemId = Guid.Empty;
-                                if (this == Globals.Me)
-                                {
-                                    var slot = MyEquipment[equipSlot];
-                                    if (slot > -1)
-                                    {
-                                        itemId = Inventory[slot].ItemId;
-                                    }
-                                }
-                                else
+                                var slot = MyEquipment[equipSlot];
+                                if (slot > -1)
                                 {
-                                    itemId = Equipment[equipSlot];
+                                    itemId = Inventory[slot].ItemId;
                                 }
+                            }
+                            else
+                            {
+                                itemId = Equipment[equipSlot];
+                            }
 
-                                var item = ItemBase.Get(itemId);
-                                if (item != null)
-                                {
-                                    if (Gender == 0)
-                                    {
-                                        DrawEquipment(item.MalePaperdoll, renderColor.A);
```

Comment:
Does "renderColor" cover this part? if you look back at this previous implementation, equipment wasn't being rendered properly with it's defined RBGA settings, but only with alpha for some reason.

Around last week, i tried to modify the DrawEquipment function to cover up this issue, by doing something like this:
![imagen](https://user-images.githubusercontent.com/17498701/144725543-00cc06ae-7f27-4696-b304-24d8a4780cf8.png)

Of course, this new logic that caches what's being rendered **is way better right now, thanks a lot for this!**, back to my question, do this PR allows it properly render equipment with their colour settings? if so, that's great! and probably should be pointed out in the PR info as well, also, not sure if there's any issue opened for this right now, but is something that may be worth mentioning.

Reply:
I believe I fixed the render colours when I found out it was broken along the way here yes. Seeing how it's cached as one image and reused it should draw the same anywhere now.